test/clj/rems/application/test_commands.clj

```diff
@@ -1844,3 +1848,195 @@
             result (commands/handle-command command application command-injections)]
         (is (= {:errors [{:type :forbidden}]} result))))))
 
+(deftest test-redact-attachments-command
+  (let [handler-remark {:type :application.command/remark
+                        :actor handler-user-id
+                        :comment "handler's remark"
+                        :attachments [{:attachment/id 1}]
+                        :public false}
+        request-review {:type :application.command/request-review
+                        :actor handler-user-id
+                        :reviewers [reviewer-user-id]
+                        :comment ""}
+        reviewer-remark {:type :application.command/remark
+                         :actor reviewer-user-id
+                         :comment "reviewer's remark"
+                         :attachments [{:attachment/id 3}]
+                         :public false}
+        request-decider {:type :application.command/request-decision
+                         :actor handler-user-id
+                         :deciders [decider-user-id]
+                         :comment ""}
+        decider-remark {:type :application.command/remark
+                        :actor decider-user-id
+                        :comment "decider's remark"
+                        :attachments [{:attachment/id 5}]
+                        :public false}]
+    (testing "fail when draft application"
+      ;; XXX: this error should be :forbidden but currently permissions are checked
+      ;; after command validation, see rems.application.commands/handle-command
+      (is (= {:errors [{:type :empty-redacted-attachments}]}
+             (fail-command {:type :application.command/redact-attachments
+                            :actor handler-user-id
+                            :comment "should fail"
+                            :public false
+                            :redacted-attachments []
+                            :attachments []}
+                           (build-application-view [dummy-created-event])))))
```

Comment:
I think the error can go either way because it's a valid error too. So shouldn't this be using a case where there is a draft with an attachment? The next case already checks that empty redacted attachment should fail. For the command to go through it needs to be otherwise valid except that it has the wrong user doing it?